packages/webview_flutter/webview_flutter_web/example/lib/web_view.dart

```diff
@@ -0,0 +1,385 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:webview_flutter_platform_interface/webview_flutter_platform_interface.dart';
+import 'package:webview_flutter_web/webview_flutter_web.dart';
+
+/// Optional callback invoked when a web view is first created. [controller] is
+/// the [WebViewController] for the created web view.
+typedef WebViewCreatedCallback = void Function(WebViewController controller);
+
+/// A web view widget for showing html content.
+///
+/// The [WebView] widget wraps around the [WebWebViewPlatform].
+///
+/// The [WebView] widget is controlled using the [WebViewController] which is
+/// provided through the `onWebViewCreated` callback.
+///
+/// In this example project it's main purpose is to facilitate integration
+/// testing of the `webview_flutter_web` package.
+class WebView extends StatefulWidget {
```

Comment:
I find it a little bit odd that an example app needs all this boilerplate, shouldn't this be provided by the core plugin? Or is this here to prevent a circular dependency?

Is this a reduced copy of https://github.com/flutter/plugins/blob/master/packages/webview_flutter/webview_flutter/lib/src/webview.dart ?

Reply:
That's correct. The other examples from the android and ios packages have a similar amount of boilerplate, as they all have a reduced copy of this class.